test/1.4/events/T_after_control.dml

```diff
@@ -0,0 +1,159 @@
+/*
+  © 2021 Intel Corporation
+  SPDX-License-Identifier: MPL-2.0
+*/
+dml 1.4;
+device test;
+import "testing.dml";
+
+template domain {
+    shared method post_single_operator_dbl_shared(bool use_cycles, double t) {
+        if (use_cycles) {
+            after cast(t, uint64) cycles: single_operator.dbl();
+        } else {
+            after t s: single_operator.dbl();
+        }
+    }
+
+    method post_single_operator_dbl_unshared(bool use_cycles, double t) {
+        if (use_cycles) {
+            after cast(t, uint64) cycles: single_operator.dbl();
+        } else {
+            after t s: single_operator.dbl();
+        }
+    }
+
+    shared method post_single_operator_modify_shared(
+        bool use_cycles, double t, int v) {
+        if (use_cycles) {
+            after cast(t, uint64) cycles: single_operator.modify(v);
+        } else {
+            after t s: single_operator.modify(v);
+        }
+    }
+
+    method post_single_operator_modify_unshared(
+        bool use_cycles, double t, int v) {
+        if (use_cycles) {
+            after cast(t, uint64) cycles: single_operator.modify(v);
+        } else {
+            after t s: single_operator.modify(v);
+        }
+    }
+
+    shared method post_multi_operator_dbl_shared(
+        bool use_cycles, double t, int i, int j) {
+        if (use_cycles) {
+            after cast(t, uint64) cycles: multi_operator[i][j].dbl();
+        } else {
+            after t s: multi_operator[i][j].dbl();
+        }
+    }
+
+    method post_multi_operator_dbl_unshared(
+        bool use_cycles, double t, int i, int j) {
+        if (use_cycles) {
+            after cast(t, uint64) cycles: multi_operator[i][j].dbl();
+        } else {
+            after t s: multi_operator[i][j].dbl();
+        }
+    }
+
+    shared method post_multi_operator_modify_shared(
+        bool use_cycles, double t, int i, int j, int v) {
+        if (use_cycles) {
+            after cast(t, uint64) cycles: multi_operator[i][j].modify(v);
+        } else {
+            after t s: multi_operator[i][j].modify(v);
+        }
+    }
+
+    method post_multi_operator_modify_unshared(
+        bool use_cycles, double t, int i, int j, int v) {
+        if (use_cycles) {
+            after cast(t, uint64) cycles: multi_operator[i][j].modify(v);
+        } else {
+            after t s: multi_operator[i][j].modify(v);
+        }
+    }
```

Comment:
A lot of dumb repetition. I can't find a way around it while still keeping the `after` statements associated with the object instantiating the template.